query/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Salesforce, Inc. All rights reserved.
+mod generated;
+
+use anyhow::{anyhow, Result};
+use std::collections::HashMap;
+
+use pdk::hl::*;
+use url::Url;
+
+use crate::generated::config::Config;
+
+/// Value to be used if the query parameter is absent from the request.
+const UNDEFINED: &str = "Undefined";
+
+/// Function that wraps the logic of the policy to simplify the error handling.
+async fn request_filter(request_state: RequestState, config: &Config) -> Flow<()> {
+    match inner_request_filter(request_state, config).await {
+        // If we could do the transformation the request reaches the backend.
+        Ok(()) => Flow::Continue(()),
+        // We send an early response otherwise.
+        Err(resp) => Flow::Break(resp),
+    }
+}
+
+/// This function extracts the query parameters from the RequestHeadersState's path and returns them
+/// as a map.
+fn query_params(state: &RequestHeadersState) -> Result<HashMap<String, String>> {
+    // Read the request's path.
+    let path = state.path();
+
+    // Create a fake url and append the path.
+    let url = Url::parse("http://fake_base").and_then(|url| url.join(path.as_str()))?;
+
+    // Extract the query pairs from the url.
+    Ok(url
+        .query_pairs()
+        .map(|(k, v)| (k.to_string(), v.to_string()))
+        .collect())
+}
+
+/// This filter takes the query parameters and forwards them as headers.
+async fn inner_request_filter(
```

Comment:
Can we remove the query params that are being moved to the headers as well? In that way we are showing how to read, and remove query params. We would still be missing the adding part though